packages/server/src/utils/workflow.ts

```diff
@@ -74,6 +77,16 @@ function processWorkflowStream(
               transformedEvent = toSourceEvent(sourceNodes);
             }
           }
+          // Handle artifact events, transform to agentStreamEvent
+          else if (artifactEvent.include(event)) {
+            const artifactAnnotation = event.data;
+            transformedEvent = agentStreamEvent.with({
+              delta: toInlineAnnotation(artifactAnnotation),
+              response: "",
+              currentAgentName: "assistant",
+              raw: artifactAnnotation,
+            });
+          }
```

Comment:
we might need this more often, how about:
```suggestion
            transformedEvent = toInlineAnnotationEvent(event)
          }
```